모임 조회 기능 추가

hyeonnkang opened the pull request:
## 관련 이슈
- close #64 

## 작업 사항
- 모임 모집글 조회 기능 추가
- 모임 기록 조회 기능 추가
- 페이징은 안되는 상태로, 단순 모임 리스트 조회되는 상태입니다.

## 참고 자료
- 참고자료 (링크 등)

## 체크리스트
- [x]  PR의 제목은 팀 내 규칙을 준수하여 알맞게 작성하였는가?
- [x]  Merge 하는 브랜치가 올바른가? (`main` branch에 실수로 PR 생성 금지)
- [x]  팀의 코딩 컨벤션을 준수하는가?
- [x]  PR과 관련없는 변경사항이 들어가지는 않았는가?
- [x]  내 코드에 대한 자기 검토가 되었는가?
- [x]  Reviewers, Assignees, Lables, Project, Milestone은 적절하게 선택하였는가?
- [x]  관련한 issue를 닫아야 하는지 점검해보고 적용했는가?

hyeonnkang committed:
feat: #64 모임 목록 조회 기능 추가 (페이징 제외)

모임 목록을 카테고리, 동네, 제목으로 검색해 조회하는 기능을 추가했습니다. 페이징 기능은 현재 되지 않는 상태입니다.

구현하는 과정에서 MeetSearch  DTO를 생성했는데 request dto는 아니고, MeetSearchDTO로 하기에는 from, toEntity 메소드를 만들지 못할 것 같아서 현재 MeetSearch로 놔두었습니다.

hyeonnkang committed:
feat: #64 모임 기록 조회 기능 구현

- MeetService 내에서 모임 모집글 메소드와 완전히 따로 구현한 상태. 추후 비슷한 코드블럭을 묶어 따로 빼낼 예정이다.

hyeonnkang committed:
Merge branch 'develop' into feature/#64-meet-find

hyeonnkang committed:
feat: #64 주석 추가 및 예외 설정

MemberService 코드 내에서 기존 MemberIdNotFoundException throw 하는 것이 적절하나 develop 브랜치를 merge 하지 않아 EntityNotFoundException 으로 처리한 곳이 있었는데, develop 브랜치를 반영하면서 예외를 적절히 처리해주었다.

hyeonnkang committed:
feat: #64 주석 추가 및 불필요한 출력문 제거

MeetService 내 모임 목록 리스트 조회 메소드의 주석을 달아주고 필요없는 print 문을 제거해주었다.

hyeonnkang committed:
Merge branch 'develop' into feature/#64-meet-find

hyeonnkang committed:
Merge branch 'develop' into feature/#64-meet-find

hyeonnkang committed:
feat: #64 모임 수정 시 모임 멤버 인원수 가져오기 추가

이전에 MeetDto의 from 메소드 파라미터가 변경되었으므로 모임 서비스의 update 메소드에서 모임 멤버인원수를 조회해 넣어주었다.

hyeonnkang committed:
docs: #64 swagger 문서 추가

모임 기록 조회 API controller 에서 스웨거 문서를 추가해주었다.

hyeonnkang committed:
Merge branch 'develop' into feature/#64-meet-find

hyeonnkang committed:
fix: #64 버그 수정

앞선 브랜치 머지에서 MeetMember Repository 의 findAllByMeet_Id 함수가 누락되어서 이 부분을 추가해주었다.

Wo-ogie commented on src/main/java/com/umc/withme/service/MeetService.java:
```diff
@@ -70,7 +72,14 @@ public MeetDto findById(Long meetId) {
+        long membersCount = meetMemberRepository.findAllByMeet_Id(meetId)
+                .stream()
+                .map(mm -> mm.getMember())
+                .count();
```
단순 개수만 필요하신거면 count하는 query를 사용하시는게 좋을 것 같아요~

hyeonnkang committed:
feat: #64 코드리뷰 반영

좋아요 수 및 모임의 멤버 수를 서비스에서 설정하는 것이 아닌 DTO 내부에서 임의로 설정하는 것으로 변경해서 서비스 코드에 영향이 덜 가도록 변경했다. 추후 좋아요 및 모임의 멤버 수가 구현이 됬을 경우 해당 부분을 변경할 예정이다.

dldmldlsy commented on src/main/java/com/umc/withme/controller/MeetController.java:
```diff
@@ -146,4 +146,66 @@ public ResponseEntity<BaseResponse> deleteMeet(
+            @RequestParam(value = "sido", required = false) String sido,
+            @RequestParam(value = "sgg", required = false) String sgg,
+            @RequestParam(value = "title", required = false) String title
+    ) {
```
이 api에서는 프론트에게 지역 이름이 아닌, 동네범위설정상태가 내동네인지 온동네인지만 전달받는 게 더 좋을 것같아요! 
프론트측에서는 사용자의 동네 이름까지 다 알고있을 수 없기때문에 기존 코드대로라면 프론트는 이 api를 요청하기 전에 사용자의 동네 이름을 조회하는 작업을 매번 거쳐야하는 번거로움이 있을 것같아서요,,!! 
동네범위를 나타내는 변수를 request param으로 두는 건 어떨까요??

dldmldlsy replied:
그리고 카테고리가 전체인 경우와 검색어가 따로 없는 경우는 해당 필드를 널값으로 보내도 된다는 설명이 있으면 좋을 것같아욥!

Wo-ogie commented on src/main/java/com/umc/withme/controller/MeetController.java:
```diff
@@ -146,4 +146,66 @@ public ResponseEntity<BaseResponse> deleteMeet(
         );
     }
+
+    /**
+     * 모임 모집글 리스트 조회 API
+     * 조건에 해당하는 모임 모집글 목록들을 조회한다.
+     *
+     * @param category 조회할 모임의 카테고리
+     * @param sido     조회할 모임의 동네 시/도 주소
+     * @param sgg      조회할 모임의 동네 시/군/구 주소
+     * @param title    조회할 모임의 제목
+     * @return 조건에 해당하는 모임 DTO 리스트를 반환한다.
+     */
```
저희 팀 문서 보시면 API는 명세 있으니까 주석 따로 안달아도 괜찮겠다는 내용이 있어서요.
지워주셔도 될 것 같습니다!

Wo-ogie commented on src/main/java/com/umc/withme/dto/meet/MeetInfoListGetResponse.java:
```diff
@@ -0,0 +1,28 @@
+    @Schema(description = "조회한 모임 정보 리스트")
+    private List<MeetInfoGetResponse> meetInfoGetResponses;
+
+    public static MeetInfoListGetResponse from(List<MeetDto> meetDtos) {
+        List<MeetInfoGetResponse> meetInfoGetResponseList = new ArrayList<>();
+        for (MeetDto meetDto : meetDtos) {
+            meetInfoGetResponseList.add(MeetInfoGetResponse.from(meetDto));
+        }
+        return new MeetInfoListGetResponse(meetInfoGetResponseList);
+    }
```
단순 dto type 변환은 java stream 쓰면 되게 깔끔해집니다!
```java
meetDtos.stream()
        .map(MeetInfoGetResponse::from)
        .collect(Collectors.toUnmodifiableList())
```

Wo-ogie commented on src/main/java/com/umc/withme/controller/MeetController.java:
```diff
@@ -146,4 +146,66 @@ public ResponseEntity<BaseResponse> deleteMeet(
+    )
+    @GetMapping("/meets")
+    public ResponseEntity<DataResponse<MeetInfoListGetResponse>> getMeets(
+            @RequestParam(value = "category", required = false) MeetCategory category,
+            @RequestParam(value = "sido", required = false) String sido,
+            @RequestParam(value = "sgg", required = false) String sgg,
+            @RequestParam(value = "title", required = false) String title
```
요청 값들에 대한 validation 처리가 있으면 더 좋을 것 같아요! Enum이나 다른건 저도 확실히 기억이 안나서 해봐야 알 것 같긴 한데,,,, 적어도 제목은 `@NotBlank`를 달아주는게 낫지 않을까 싶어요. 필수값이 아닌거랑 상관없이, 만약 전달되었을 때 빈 문자열이면 안되는 값이니까요.
혹시 다른 parameter들도 테스트 해보시고 할만한 게 있다면 validation 처리 해주시면 좋을 것 같습니다.
참고로 `@RequestParam`에 validation을 적용하려면 class level에 `@Validated`를 붙이고 각 parameter에 적용할 validation annotation을 적어주면 되는 것 같더라구요?

Wo-ogie commented on src/main/java/com/umc/withme/dto/meet/MeetSearch.java:
```diff
@@ -0,0 +1,38 @@
+    private MeetStatus meetStatus;  // 모임 진행상태
+    private Long memberId;  // 사용자 id
+
+    public static MeetSearch of(MeetCategory category, String sido, String sgg, String title) {
+        return new MeetSearch(
+                category != null ? category : null,
+                sido != null && sgg != null ? AddressRequest.of(sido, sgg) : null,
+                title,
+                null,
+                null);
+    }
+
+    public static MeetSearch of(MeetStatus meetStatus, Long memberId) {
+        return new MeetSearch(null,
+                null,
+                null,
+                meetStatus,
+                memberId);
+    }
```
서로 사용하는 필드가 겹치는게 전혀 없다면 모임 조회와 모임 기록 조회의 request를 분리하는 것도 좋을 것 같습니다.

hyeonnkang committed:
Merge branch 'develop' into feature/#64-meet-find

hyeonnkang committed:
feat: #64 MeetDto from 메소드 파라미터 추가

from 메소드에 파라미터 2개가 비어있어 이를 추가해주었다.

hyeonnkang committed:
feat: #64 코드리뷰 반영

hyeonnkang committed:
feat: #64 코드리뷰 반영

- 모임 모집글 조회 및 모임 기록 조회 시 검색 DTO, repository method 분리
- 모임 모집글 조회 시 title 검색 조건에 문자열의 길이가 2 이상이 되도록 추가

hyeonnkang committed:
feat: #64 코드리뷰 반영

치즈의 리뷰를 반영하여 모임 모집글 목록 api 의 request 중 주소를 받는 부분을 변경하였습니다.
